ad-joining/register-computer/join.ps1

```diff
@@ -110,5 +158,30 @@ if ($JoinInfo) {
                     "Joining computer to domain failed: $($_.Exception.Message)")
             }
         }
+        finally
```

Comment:
You're still in the do/while loop here -- shouldn't that cleanup logic be at the very end?

Reply:
Yes it should. This works as long as the join completes the first time. Good catch.